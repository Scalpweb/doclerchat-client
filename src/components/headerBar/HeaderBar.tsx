import React from 'react';
import IconButton from 'components/iconButton/IconButton';
import styles from './HeaderBar.module.scss';

interface Props {
  showSettingsButton: boolean;
  showChatButton: boolean;
  onClickSettings: () => void;
  onClickChat: () => void;
  unreadCount?: number;
}

const HeaderBar = (props: Props) => {
  return (
    <div className={styles.headerBar}>
      <div className={styles.title}>Docler Chat</div>
      {props.showChatButton && (
        <div className="pull-left">
          <IconButton onClick={props.onClickChat} icon="chat" counter={props.unreadCount || 0} />
        </div>
      )}
      {props.showSettingsButton && (
        <div className="pull-right">
          <IconButton onClick={props.onClickSettings} icon="settings" />
        </div>
      )}
    </div>
  );
};

export default HeaderBar;
