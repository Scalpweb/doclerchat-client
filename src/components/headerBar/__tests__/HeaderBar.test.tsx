import React from 'react';
import {shallow} from 'enzyme';
import HeaderBar from '../HeaderBar';

const emptyFnc = () => {
  return;
};

describe('HeaderBar', () => {
  it('renders without button', () => {
    const component = shallow(
      <HeaderBar
        onClickChat={emptyFnc}
        onClickSettings={emptyFnc}
        showSettingsButton={false}
        showChatButton={false}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders buttons', () => {
    const component = shallow(
      <HeaderBar
        onClickChat={emptyFnc}
        onClickSettings={emptyFnc}
        showSettingsButton={true}
        showChatButton={true}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders unread counter', () => {
    const component = shallow(
      <HeaderBar
        onClickChat={emptyFnc}
        onClickSettings={emptyFnc}
        showSettingsButton={false}
        showChatButton={true}
        unreadCount={1}
      />
    );
    expect(component).toMatchSnapshot();

    component.setProps({unreadCount: 0});
    expect(component).toMatchSnapshot();
  });
});
