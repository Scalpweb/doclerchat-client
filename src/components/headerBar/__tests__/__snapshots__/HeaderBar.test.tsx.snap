// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar renders buttons 1`] = `
<div>
  <div>
    Docler Chat
  </div>
  <div
    className="pull-left"
  >
    <IconButton
      counter={0}
      icon="chat"
      onClick={[Function]}
    />
  </div>
  <div
    className="pull-right"
  >
    <IconButton
      icon="settings"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`HeaderBar renders unread counter 1`] = `
<div>
  <div>
    Docler Chat
  </div>
  <div
    className="pull-left"
  >
    <IconButton
      counter={1}
      icon="chat"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`HeaderBar renders unread counter 2`] = `
<div>
  <div>
    Docler Chat
  </div>
  <div
    className="pull-left"
  >
    <IconButton
      counter={0}
      icon="chat"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`HeaderBar renders without button 1`] = `
<div>
  <div>
    Docler Chat
  </div>
</div>
`;
